src/internal/pfsdb/commits.go

```diff
@@ -963,6 +963,18 @@ func ForEachCommitTxByFilter(ctx context.Context, tx *pachsql.Tx, filter *pfs.Co
 	return nil
 }
 
+func ListCommitWithIdTxByFilter(ctx context.Context, tx *pachsql.Tx, filter *pfs.Commit, orderBys ...OrderByCommitColumn) ([]*CommitWithID, error) {
```

Comment:
This function name is a mouthful, there is probably a better way to compose this functionality. I'm fine with this for now though because this seems to be the general pattern for these utility functions. As a side note, I wonder if we would be better off just having `pfsdb.Commit` rather than `pfsdb.CommitWithID`. That seems more readable, and the packaging of `src/pfs` vs `src/internal/pfsdb` would disambiguate between the external and internal commit abstractions.

Reply:
I'll do that as a follow up